float32 lat1
float32 lng1
float32 lat2
float32 lng2
float32 lat3
float32 lng3
float32 lat4
float32 lng4
